### ISSUE TITLE
Left nav pane (Origin: bugzilla #774167)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#6107]
status NEEDINFO severity *normal* in component *general* for *---*
Reported in version *1.8.11* on platform *Other*
Assigned to: Dimitri van Heesch

Original attachment names and IDs:
- _[ExampleCppCodeB774167-skewed.zip](https://sourceforge.net/projects/doxygen/files/attachments/bug774167_343672_ExampleCppCodeB774167-skewed.zip/download/download)_ (ID 343672)

On 2016-11-09 21:31:55 +0000, Bob D. wrote:

    When we modify the doxygenlayout.xml to hide some panes, they are hidden in
    the top pane, but they could remain visible in the left nav pane, which is
    not the desired behaviour. Too bad...

On 2016-11-10 20:50:13 +0000, Dimitri van Heesch wrote:

    Do I understand your request correctly that you want to configure the tabs
    and the navigation pane independently?
    
    Or is the problem that you hid a tab in the top pane, but then still see the
    corresponding entry in the navigation pane?

On 2016-11-10 21:48:15 +0000, Bob D. wrote:

    Yep, sorry, I was not clear.
    First, some explanations: I am using doxygen to parse C++ code and generate
    HTML documentation.
    
    What I meant was that, in the doxygenlayout.xml, you can some entries by
    using the 'visible="no"' attribute, in the <navindex> node. But I tried to
    remove, using 'visible="no"' on the "files" node. But the "filelist" and
    "globals" node, in the files node, came out at an upper level. I figured how
    to make them disappear in both menus: by setting visible="no" to every
    subnodes.
    
    I do not try to configure the left nav pane and the tabs independently. But
    the problem was indeed that I got different results that I couldn't explain.
    
    But I notice another bug: when I work on a project with no namespaces, but
    having the namespaces set to visible, I don't see it in the tabs, but I see
    it in the left pane. Besides, it does find something for the namespaces, but
    coming from the INCLUDE_PATH. Anyway, we don't have the same result in the
    two panes, which is surprising. <---- Second bug.

On 2016-11-11 09:15:05 +0000, Dimitri van Heesch wrote:

    Can you please attach a self-contained example (source+config file in a tar
    or zip) that allows me to reproduce the problem(s)?

On 2017-01-17 20:34:21 +0000, Bob D. wrote:

    Created attachment 343672
    Example with a problem on the left pane. The file list category was exected
    to be hidden.
    
    Example with a problem on the left pane. The file list category was exected
    to be hidden.

**Comment by albert-github** [comment]
The relevant code in the `DoxygenLayout.xml` is:
```
    <tab type="files" visible="no" title="">
      <tab type="filelist" visible="no" title="" intro=""/>
      <tab type="globals" title="" intro=""/>
    </tab>
```
It is expected that the sub-tabs `filelist` and `globals` would disappear in the treeview as soon as the top tab `files` is set to visible `no` as it happens in the 'menubar'. This is not the case, a workaround is to set the visibility explicitly to `no`:
```
    <tab type="files" visible="no" title="">
      <tab type="filelist" visible="no" title="" intro=""/>
      <tab type="globals" visible="no" title="" intro=""/>
    </tab>
```

**Comment by albert-github** [comment]
Slightly adjusted code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
